### PR TITLE
Enable UPerf to run multiple tests

### LINKED ISSUE #92: Multiple UPerf tests w/ single CR
We want to have a single UPerf CR that iterates through multiple UPerf scenarios, for example:

```
apiVersion: benchmark.example.com/v1alpha1
kind: Benchmark
metadata:
  name: uperf-benchmark
  namespace: ripsaw
spec:
  workload:
    # cleanup: true
    name: uperf
    args:
      hostnetwork: true 
      pin: true 
      pin_server: "master-0"
      pin_client: "master-1"
      rerun: 1 
      pair: 1
      protos: 
        - tcp
      test_type: stream
      nthr: 2
      sizes: 
        - 16384
        - 1024
      runtime: 30
```

The above test will iterate through 16384 and 1024 message sizes. 

There are two approaches I see, one of which I have implemented and tested.

### First Option
Build UPerf XML to iterate through the tests.

However, the output isn't obvious, meaning:
```
TX worklist success  Sent workorder
Handshake phase 2 with 192.168.111.20 done
Completed handshake phase 2
Starting 2 threads running profile:ripsaw-test ...   0.00 seconds
TX command [UPERF_CMD_NEXT_TXN, 0] to 192.168.111.20
Txn1          0 /   0.00(s) =            0           0op/s 
Txn1          0 /   1.00(s) =            0           2op/s 

TX command [UPERF_CMD_NEXT_TXN, 1] to 192.168.111.20
Txn2          0 /   0.00(s) =            0           0op/s 
Txn2     2.75GB /   1.00(s) =    23.57Gb/s      179816op/s 
Txn2     5.49GB /   2.00(s) =    23.55Gb/s      179686op/s 
Txn2     8.23GB /   3.00(s) =    23.54Gb/s      179623op/s 
Txn2    10.97GB /   4.00(s) =    23.54Gb/s      179597op/s 
Txn2    13.71GB /   5.00(s) =    23.54Gb/s      179592op/s 
Txn2    16.46GB /   6.01(s) =    23.54Gb/s      179575op/s 
Txn2    19.20GB /   7.01(s) =    23.54Gb/s      179566op/s 
Txn2    21.94GB /   8.01(s) =    23.53Gb/s      179551op/s 
Txn2    24.68GB /   9.01(s) =    23.53Gb/s      179549op/s 
Txn2    27.39GB /  10.01(s) =    23.50Gb/s      179310op/s 
Txn2    30.13GB /  11.01(s) =    23.51Gb/s      179335op/s 
Txn2    32.87GB /  12.01(s) =    23.51Gb/s      179357op/s 
Txn2    35.61GB /  13.01(s) =    23.51Gb/s      179362op/s 
Txn2    38.36GB /  14.01(s) =    23.51Gb/s      179383op/s 
Txn2    41.10GB /  15.01(s) =    23.51Gb/s      179392op/s 
Txn2    43.83GB /  16.02(s) =    23.51Gb/s      179359op/s 
Txn2    46.58GB /  17.02(s) =    23.51Gb/s      179373op/s 
Txn2    49.32GB /  18.02(s) =    23.51Gb/s      179378op/s 
Txn2    52.06GB /  19.02(s) =    23.51Gb/s      179391op/s 
Txn2    54.80GB /  20.02(s) =    23.51Gb/s      179397op/s 
Txn2    57.54GB /  21.02(s) =    23.51Gb/s      179400op/s 
Txn2    60.28GB /  22.02(s) =    23.52Gb/s      179409op/s 
Txn2    63.03GB /  23.02(s) =    23.52Gb/s      179417op/s 
Txn2    65.77GB /  24.02(s) =    23.52Gb/s      179418op/s 
Txn2    68.51GB /  25.02(s) =    23.52Gb/s      179424op/s 
Txn2    71.25GB /  26.02(s) =    23.52Gb/s      179429op/s 
Txn2    73.99GB /  27.03(s) =    23.52Gb/s      179428op/s 
Txn2    76.70GB /  28.03(s) =    23.51Gb/s      179357op/s 
Txn2    79.45GB /  29.03(s) =    23.51Gb/s      179365op/s 
Sending signal SIGUSR2 to 140611952076544
Sending signal SIGUSR2 to 140611943683840
called out
Txn2    82.19GB /  30.23(s) =    23.35Gb/s      178180op/s 

TX command [UPERF_CMD_NEXT_TXN, 2] to 192.168.111.20
Txn3          0 /   0.00(s) =            0           0op/s 
Txn3          0 /   1.00(s) =            0           2op/s 

TX command [UPERF_CMD_NEXT_TXN, 3] to 192.168.111.20
Txn4          0 /   0.00(s) =            0           0op/s 
Txn4          0 /   1.00(s) =            0           2op/s 

TX command [UPERF_CMD_NEXT_TXN, 4] to 192.168.111.20
Txn5          0 /   0.00(s) =            0           0op/s 
Txn5     1.61GB /   1.00(s) =    13.80Gb/s     1685114op/s 
Txn5     3.23GB /   2.00(s) =    13.85Gb/s     1690938op/s 
Txn5     4.81GB /   3.00(s) =    13.77Gb/s     1681184op/s 
Txn5     6.47GB /   4.00(s) =    13.88Gb/s     1694944op/s 
Txn5     8.09GB /   5.00(s) =    13.89Gb/s     1695700op/s 
Txn5     9.73GB /   6.01(s) =    13.91Gb/s     1698501op/s 
Txn5    11.32GB /   7.01(s) =    13.88Gb/s     1694143op/s 
Txn5    12.90GB /   8.01(s) =    13.84Gb/s     1689175op/s 
Txn5    14.52GB /   9.01(s) =    13.85Gb/s     1690120op/s 
Txn5    16.14GB /  10.01(s) =    13.85Gb/s     1690853op/s 
Txn5    17.76GB /  11.01(s) =    13.86Gb/s     1691405op/s 
Txn5    19.41GB /  12.01(s) =    13.88Gb/s     1694451op/s 
Txn5    21.04GB /  13.01(s) =    13.89Gb/s     1695563op/s 
Txn5    22.67GB /  14.01(s) =    13.90Gb/s     1696544op/s 
Txn5    24.28GB /  15.01(s) =    13.89Gb/s     1695637op/s 
Txn5    25.91GB /  16.02(s) =    13.89Gb/s     1696059op/s 
Txn5    27.53GB /  17.02(s) =    13.90Gb/s     1696469op/s 
Txn5    29.18GB /  18.02(s) =    13.91Gb/s     1698319op/s 
Txn5    30.82GB /  19.02(s) =    13.92Gb/s     1699048op/s 
Txn5    32.46GB /  20.02(s) =    13.93Gb/s     1700361op/s 
Txn5    34.10GB /  21.02(s) =    13.93Gb/s     1700785op/s 
Txn5    35.68GB /  22.02(s) =    13.92Gb/s     1699024op/s 
Txn5    37.29GB /  23.02(s) =    13.91Gb/s     1698154op/s 
Txn5    38.93GB /  24.02(s) =    13.92Gb/s     1699317op/s 
Txn5    40.57GB /  25.03(s) =    13.93Gb/s     1699966op/s 
Txn5    42.21GB /  26.03(s) =    13.93Gb/s     1700781op/s 
Txn5    43.86GB /  27.03(s) =    13.94Gb/s     1701440op/s 
Txn5    45.46GB /  28.03(s) =    13.93Gb/s     1700898op/s 
Txn5    47.10GB /  29.03(s) =    13.94Gb/s     1701272op/s 
Sending signal SIGUSR2 to 140611952076544
Sending signal SIGUSR2 to 140611943683840
```

It isn't obvious to the reader which test is what -- (I know what is what, but to a user, this isn't too obvious). 

### Option 2
Instead of having UPerf iterate through the tests, create a unique client for each workload. 

I personally am preferring Option 2, but want to get feedback before I go down the route of implementing this...

### PR DESCRIPTION
**Pull request by jtaleric**
This commit allows the benchmark runner to pass a lists of scenarios to
iterate through.

Fixes #92

### CODE REVIEW

File: docs/uperf.md [hunk 1]
```diff
@@ -18,19 +18,24 @@ metadata:
...
-    test_type: stream
-    nthr: 2
-    size: 16384
-    runtime: 60
+  workload:
+    # cleanup: true
+    name: uperf
+    args:
+      hostnetwork: false
+      pin: false
```
Comment (portante) [inline docs/uperf.md]: Why change `pin` argument?

File: docs/uperf.md [hunk 2]
```diff
@@ -18,19 +18,24 @@ metadata:
...
+      pin_server: "master-0"
+      pin_client: "master-1"
+      rerun: 1
```
Comment (portante) [inline docs/uperf.md]: Do you document what `rerun` means?

File: docs/uperf.md [hunk 3]
```diff
@@ -18,19 +18,24 @@ metadata:
...
+      pair: 1
+      nthr: 2
```
Comment (portante) [inline docs/uperf.md]: The `nthr` means "number of threads"?  Is this name, `nthr`, what uperf users?  Or just a short hand?

File: resources/crds/benchmark_v1alpha1_uperf_cr.yaml [hunk 4]
```diff
@@ -10,12 +10,15 @@ spec:
     args:
       hostnetwork: false
       pin: false
-      pin_server: "server_hostname"
-      pin_client: "client_hostname"
+      pin_server: "master-0"
```
Comment (portante) [inline resources/crds/benchmark_v1alpha1_uperf_cr.yaml]: Any particular reason for `master-0` and `master-1`?  Are they workable default values?

File: docs/uperf.md (hunk 3 above)
Comment (portante) [reply docs/uperf.md]: Maybe rename it to match the `group.nthreads` in the uperf yml?

File: resources/crds/benchmark_v1alpha1_uperf_cr.yaml (hunk 4 above)
Comment (jtaleric) [reply resources/crds/benchmark_v1alpha1_uperf_cr.yaml]: Ah, no they are not -- this is my CR I have used for testing :)

File: docs/uperf.md (hunk 3 above)
Comment (jtaleric) [reply docs/uperf.md]: yup, took `nthr` straight out of uperf.

File: docs/uperf.md (hunk 2 above)
Comment (jtaleric) [reply docs/uperf.md]: Oh, good point I do not. I will add that!

File: docs/uperf.md (hunk 1 above)
Comment (jtaleric) [reply docs/uperf.md]: For documentation sake I just copied my previous CR. We could leave it enabled for the docs.

**Pull request merged**